### DISCUSSION TITLE
Immediately apply filter to new or modified nodes on unselected views

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#1950] [Ideas]
Related feature request: https://github.com/freeplane/freeplane/issues/1946

---
As explained in the feature request above, the question is:

What should happen when:

- the user has two map views
- one of the map views (e.g., left map view) has a filter applied
- on the right map view, the user makes a change in a node, that makes the node become outside the scope of the filter (or vice versa)
- so, in the left view, that node was showing up, but, now, as it is outside the scope of the filter, should it disappear (be filtered)?

**Comment by dpolivaev** [comment]
Could you please explain your concrete use case with some close to reality example?

**Reply by euu2021** [reply]
Basically, it's still the same use case that I explained 2 years ago (https://github.com/freeplane/freeplane/discussions/401#discussioncomment-2845218):

> When I want to understand a topic, I read several sources of information and collect relevant quotes. As my comprehension grows, I rearrange the branches to reflect my overall view of the topic. The problem of this approach is scalability: after some time I have several pages worth of content, which makes it hard to insert the quotes on proper places and rearrange nodes and branches. Even in Freeplane this scalability problem happens.

So, I have to, non-linearly, insert information along a branch with hundreds or thousands of nodes. This gets progressively more difficult as the branch grows, because I need to navigate a giant branch in order to find the proper spot to insert the information.

Now, the situation improved, because, with the new feature `Navigate->Synchronize selection`, one more thing that I can do is arranging the information in categories nodes (with the style "structure"), and then keep a second map view, showing only the "structure" nodes (by using a filter), which makes the navigation easier.

So, you can see that I want the "structure" map view to have only "structure" nodes. Thus, I want that, as I work on the main map view, the "structure" map view, keeps automatically updated to show only "structure" nodes.

**Comment by aaa1386** [comment]
FP should update the left view. This function is very logical. Because that view is actually set to filter a certain style, so it should always have the number of nodes it shows equal to the actual number of nodes that have that style.
Basically, in some cases, two views should be completely independent from each other. Like the Auto-Expand option.
Sometimes they should not act independently, but the changes in each view should be updated in the other view, like the case mentioned on this page. Another example is the "Scroll map after selecting a node" option. That is, if the user selects a node in the right view, in the left view that node is automatically selected by FP, the map should be scrolled automatically.
There is no such function now.

Sometimes it is possible to define for the software that the update is done from one view to another but not the other way around. This makes the user have two types of functions between the two views. I stated one of the cases that it is better to be this way at : https://github.com/freeplane/freeplane/discussions/1936#discussioncomment-10192857
In order for FP to understand such a situation, the first view that opens and the subsequent views must be defined separately for FP. For example, let's say to FP: the first time the user opens a map, think of it as the main map. If the user opens another view of that map, consider it as an auxiliary view.  Then you can define for FP what function each of the views should have.

**Comment by dpolivaev** [comment]
Preview 1.12.02_15
